CMake/coverage.cmake

```diff
@@ -35,8 +35,46 @@ function(add_converage module)
         add_custom_command(TARGET ${module}-ccov POST_BUILD
             COMMENT "Open ${module}-llvm-cov/index.html in your browser to view the coverage report."
         )
+    elseif("${CMAKE_C_COMPILER_ID}" MATCHES "GNU" OR "${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
+        message("[Coverage] Building with Gcc Code Coverage Tools")
+
+        find_program(GCOV_PATH gcov)
+        IF(NOT GCOV_PATH)
+	       MESSAGE(FATAL_ERROR "gcov not found! Aborting...")
+        ENDIF() # NOT GCOV_PATH
+        find_program(GCOVR_PATH gcovr)
+        IF(NOT GCOVR_PATH)
+	       MESSAGE(FATAL_ERROR "gcovr not found! Aborting...")
```

Comment:
Some formatting could be nice in the file. The tab spaces are not equal on different lines like this one.